packages/messaging/src/ui/components/UnreadIcon/UnreadIcon.tsx

```diff
@@ -1,11 +1,223 @@
-import { Text, View } from "react-native";
+/*
+    Copyright 2025 Adobe. All rights reserved.
+    This file is licensed to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable law
+    or agreed to in writing, software distributed under the License is
+    distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS OF
+    ANY KIND, either express or implied. See the License for the specific
+    language governing permissions and limitations under the License.
+*/
+import React, { useState } from 'react';
+import {
+  Image,
+  ImageProps,
+  ImageStyle,
+  StyleSheet,
+  View,
+  ViewProps,
+  ViewStyle,
+  useColorScheme
+} from 'react-native';
+import useContainerSettings from '../../hooks/useContainerSettings';
+
+export interface UnreadIconProps extends ViewProps {
+  imageStyle?: ImageStyle;
+  containerStyle?: ViewStyle;
+  source?: ImageProps['source'];
+  darkSource?: ImageProps['source'];
+  size?: number;
+  position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+  type?: 'dot' | 'image';
+}
+
+// Helper function to convert placement from settings to component position
+const convertPlacement = (placement: 'topleft' | 'topright' | 'bottomleft' | 'bottomright'): 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' => {
```

Comment:
not sure we really need this, also Id create a type for the placement if you want to keep it